advice_for_future_giraf/process_advice.md

```diff
@@ -0,0 +1,38 @@
+Be careful to consider the following points when you think out the process for your own project:
+
+* How do you find dependencies between user stories? 
+    + How do you handle dependencies once discovered?
+* How do you make sure that one group's work is not made obsolete by another's? (Especially when the two groups are working in the same files.)
+
+We found that handling dependencies needs constant work with our process. It is quickly forgotten by individual groups that their work might affect others or that others might depend on one user story to implement their own if people are not made aware of this constantly.
+
+## Do Mention Everything Explicitly
+Throughout the 2019 Giraf project we found that the only way to make sure that every group followed the process correctly was to tell them every part of it (sometimes repeatedly). While this might sound trivial, there are a lot of small things that can be seen as implicit to some people and if the process group finds something to be self-evident they might not mention it. However small any part of the process might be, it should always be mentioned explicitly to everyone in the project.<br>
+If you decide to use some form of frequent meetings (e.g. daily Scrum of Scrums stand up meetings) this is a good place to remind people of small things related to process. Even if it might be redundant at the end, at least you know that people are on the same page.
+
+## Do Plan All Meetings of a Sprint at the Beginning of it
+It is important that everyone on the Giraf project knows when certain milestones and meetings occur during a sprint. It is all too easy to schedule these (especially frequent recurring meetings such as Scrum of Scrums stand up meetings) on a week by week basis. However, we found that with this way of scheduling, some meetings were scheduled too late for people to notice. This caused some stress for the different groups as they could not schedule their work around meetings far in advance. Later in the project we began scheduling all meetings of a sprint at the beginning of the sprint itself. This meant that everyone knew when all meetings were happening and it also meant that no meetings had to be delayed or skipped because of annnouncing them too late.
+
+## Do Enforce a Certain Standard for the Code
+For most people this will be the first time that they work on a project like this. While many might have different ways of writing code it is important that everyone adheres to same same standards for this project. Since the project is an ongoing one that is being developed by many different people through many semesters it is important to write understandable code. It is all too easy to just code away to maximize what you achieve during a semester, but if the people after you do not understand the code they might not even use it.
```

Comment:
```suggestion
For most people this will be the first time that they work on a project like this. While many might have different ways of writing code it is important that everyone adheres to same standards for this project. Since the project is an ongoing one that is being developed by many different people through many semesters it is important to write understandable code. It is all too easy to just code away to maximize what you achieve during a semester, but if the people after you do not understand the code they might not even use it.
```